backend/helpers/helpers.go

```diff
@@ -46,20 +47,16 @@ func DBErrorHandling(error error, w http.ResponseWriter) {
 	}
 }
 
-func JsonWriter(w http.ResponseWriter, table interface{}) {
-	errJson := json.NewEncoder(w).Encode(&table)
-	if errJson != nil {
-		http.Error(w, "Unable to encode response", http.StatusInternalServerError)
-	}
-}
-
-func JsonWriterString(w http.ResponseWriter, message string) {
-	messageStruct := struct {
-		Message string `json:"message"`
-	}{
-		Message: message,
+func JsonWriter(w http.ResponseWriter, response interface{}) {
+	log.Println(reflect.TypeOf(response))
+	if reflect.TypeOf(response).Kind() == reflect.String {
+		response = struct {
+			Message string `json:"message"`
+		}{
+			Message: response.(string),
+		}
 	}
-	errJson := json.NewEncoder(w).Encode(&messageStruct)
+	errJson := json.NewEncoder(w).Encode(&response)
 	if errJson != nil {
 		http.Error(w, "Unable to encode response", http.StatusInternalServerError)
 	}
```

Comment:
Reflection is very slow and creates the potential for runtime errors (the type is evaluated dynamically).
Reflection in Go is generally discouraged unless it's necessary, so I don't think it's worth introducing that behavior to save a few lines.

Reply:
there is no potential for runtime errors here

Reply:
I know, not in this case. I was speaking about reflection in general.
I just prefer to discourage unnecessary and inefficient dynamic behavior. This type of pattern is not very idiomatic for Go in my opinion.
I'll leave the final call to @AdrianJendo.